lib/urlwatch/jobs.py

```diff
@@ -284,6 +287,11 @@ def retrieve(self, job_state):
         # Save ETag from response into job_state, which will be saved in cache
         job_state.etag = response.headers.get('ETag')
 
+        # If 'pdf2text' filter is selected, return bytes content as opposed to str
+        # as it's required by the library used by that filter
+        if 'pdf2text' in self.filter or any('pdf2text' in subfilter for subfilter in self.filter):
```

Comment:
Can you turn this expression (`'pdf2text' in self.filter or any('pdf2text' in subfilter for subfilter in self.filter)`) into a helper function like `self.needs_binary_data()`? Or maybe we could make it so that this job has a "binary: true" flag or something? IMHO it would be better to do it this way (or make it so that the filter object has a "needs_binary = True" or something and find it out this way -- right now it's a bit hard to maintain).